parsers/ELEXON.py

```diff
@@ -124,337 +233,144 @@ def query_additional_eso_data(
-            if fields[column_index] == "":
-                continue  # no value provided for this exchange
-            net_flow += float(fields[column_index]) * multiplier
-        data["netFlow"] = net_flow
-        data_points.append(data)
-
-    return data_points
+    return fuelhh_data
 
 
-def parse_production_FUELINST(
-    csv_data: str,
-    target_datetime: datetime | None = None,
-    logger: Logger = getLogger(__name__),
-) -> pd.DataFrame:
-    """A temporary parser for the FUELINST report.
-    This report will be decomissioned sometime in 2023.
-    We use it as a replacement for B1620 that doesn't work
-    at the moment (19/12/2022).
-    """
-    if not csv_data:
-        raise ParserException("ELEXON.py", "Production file is empty.")
-    report = REPORT_META["FUELINST"]
-    # create DataFrame from slice of CSV rows
-    df = pd.read_csv(StringIO(csv_data), skiprows=1, skipfooter=1, header=None)
-    # check field count in report is as expected
-    field_count = len(df.columns)
-    if field_count != report["expected_fields"]:
-        raise ParserException(
-            "ELEXON.py",
-            "Expected {} fields in FUELINST report, got {}".format(
-                report["expected_fields"], len(df.columns)
-            ),
-        )
-    # The file doesn't have a column header, so we need to recreate it.
-    mapping = {1: "Settlement Date", 2: "Settlement Period", 3: "Spot Time"}
-    for index, fuel in enumerate(FUEL_INST_MAPPING.values()):
-        mapping[index + 4] = fuel
-    df = df.rename(columns=mapping)
-    df["Settlement Date"] = df["Settlement Date"].apply(
-        lambda x: datetime.strptime(str(x), "%Y%m%d")
-    )
-    df["Settlement Period"] = df["Settlement Period"].astype(int)
-    df["datetime"] = df.apply(
-        lambda x: datetime_from_date_sp(x["Settlement Date"], x["Settlement Period"]),
-        axis=1,
-    )
-    return df.set_index("datetime")
-
+def query_and_merge_production_fuelhh_and_eso(
+    session: Session, target_datetime: datetime, logger: Logger
+):
+    events_fuelhh = query_production_fuelhh(session, target_datetime, logger)
+    parsed_events_fuelhh = parse_production(events_fuelhh, logger, "FUELHH")
+    events_eso = query_additional_eso_data(target_datetime, session, logger)
+    parsed_events_eso = parse_eso_production(events_eso, logger)
 
-def parse_additional_eso_production(raw_data: list[dict]) -> pd.DataFrame:
-    """Parse additional eso data for embedded wind/solar and hydro storage."""
-    df = pd.DataFrame.from_records(raw_data)
-    df["datetime"] = df.apply(
-        lambda x: datetime_from_date_sp(x["SETTLEMENT_DATE"], x["SETTLEMENT_PERIOD"]),
-        axis=1,
+    merged_events = ProductionBreakdownList.merge_production_breakdowns(
+        [parsed_events_fuelhh, parsed_events_eso], logger
     )
-    df = df.rename(columns=ESO_FUEL_MAPPING)
-    return df.set_index("datetime")
+    return merged_events.to_list()
```

Comment:
This should not return a list, but the merged events directly, it should also check that all datetimes are mathcing.

Reply:
it should actually return a list because it will not be merged with anything else